### PR TITLE
Feature/pred2bq executor refactor

### LINKED ISSUE #221: _parse_example error when input contains SparseTensor in predictions-to-bigquery component.
## Expected Behavior
executor._parse_example should be able to parse tf.sparse.SparseTensor values that are derived from the input serialized TFRecord data.

## Actual Behavior
executor._parse_example raises an error when parsing a tf.sparse.SparseTensor value due to:
```
value_list = tensor.numpy().tolist()
```
where tensor is expected to be tf.Tensor.

## Steps to Reproduce the Problem

1. Run executor._parse_example using a serialized tf.Example proto with tf.sparse.SparseTensor value.

### PR DESCRIPTION
**Pull request by cfezequiel**
Fixes #221 

- Refactors executor.py
- Adds unit tests
- Adds credits to original code author

- [ X ] Tests pass

### CODE REVIEW

File: data [hunk 1]
```diff
@@ -0,0 +1 @@
+/usr/local/google/home/cezequiel/datasets/pred2bq
```
Comment (hanneshapke) [inline data:1]: @cfezequiel What's the use case of this file?

Comment (cfezequiel) [reply data:1]: @hanneshapke it's a symlink to a folder containing the Penguin dataset files and TFX component output files that I was using for testing. It shouldn't be in the repo. Will remove it. Thanks for catching.

**Comment by cfezequiel** [comment]
@hanneshapke @rcrowe-google if the PR looks good I was wondering if we could already merge it. I'd like to raise a subsequent PR for further code changes to the component.

**Comment by hanneshapke** [comment]
@cfezequiel I am sorry, I won't have a chance to review the PR in detail. A brief review looked good, but the PR is a bigger rewrite than what I can handle after work at the moment. I am sorry.

**Comment by hanneshapke** [comment]
@cfezequiel Just did another pass over the PR. Could you please remove my email address from the files. Can you leave my name and add my  affiliation please (e.g. `Hannes Hapke (Digits Financial Inc.)`)? Thank you.

**Comment by cfezequiel** [comment]
> @cfezequiel Just did another pass over the PR. Could you please remove my email address from the files. Can you leave my name and add my affiliation please (e.g. `Hannes Hapke (Digits Financial Inc.)`)? Thank you.
> 
> Happy to stamp / merge afterward.

@hanneshapke Sure, done.

**Comment by hanneshapke** [comment]
@cfezequiel Thank you.

**Comment by cfezequiel** [comment]
Thanks for the approval and merge @hanneshapke .

**Pull request merged**